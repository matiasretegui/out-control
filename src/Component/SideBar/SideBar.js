import React from 'react'
import { NavLink } from 'react-router-dom'

const SideBar = () =>
  <div className="bar">
    <ul>
      <li>
        <NavLink exact to='/' activeClassName="active">Balance</NavLink>
      </li>
      <li>
        <NavLink to='/gastos' activeClassName="active">Gastos</NavLink>
      </li>
      <li>
        <NavLink to='/ingresos' activeClassName="active">Ingresos</NavLink>
      </li>
      <li>
        <NavLink to="#about">About</NavLink>
      </li>
    </ul>
  </div>

export default SideBar