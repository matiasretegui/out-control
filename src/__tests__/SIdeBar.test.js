import React from 'react';
import { shallow, render, mount } from 'enzyme';
import SideBar from '../Component/SideBar/SideBar';
import { MemoryRouter } from 'react-router';


describe('SideBar', () => {
  it('should be defined', () => {
    expect(SideBar).toBeDefined();
  });

  it('should render correctly', () => {
    const tree = shallow(<SideBar />);
    expect(tree).toMatchSnapshot();
  });

  it('should have an active class on home link', () => {
    const tree = mount(
      <MemoryRouter initialEntries={[ '/' ]}>
        <SideBar/>
      </MemoryRouter>
    );
    expect(tree.find('.active').at(1).text()).toEqual('Balance')

  });
});