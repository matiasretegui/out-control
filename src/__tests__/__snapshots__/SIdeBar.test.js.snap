// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SideBar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul>
        <li>
                <NavLink
                        activeClassName="active"
                        aria-current="page"
                        exact={true}
                        to="/"
                >
                        Balance
                </NavLink>
        </li>
        <li>
                <NavLink
                        activeClassName="active"
                        aria-current="page"
                        to="/gastos"
                >
                        Gastos
                </NavLink>
        </li>
        <li>
                <NavLink
                        activeClassName="active"
                        aria-current="page"
                        to="/ingresos"
                >
                        Ingresos
                </NavLink>
        </li>
        <li>
                <NavLink
                        activeClassName="active"
                        aria-current="page"
                        to="#about"
                >
                        About
                </NavLink>
        </li>
</ul>,
      "className": "bar",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li>
            <NavLink
                        activeClassName="active"
                        aria-current="page"
                        exact={true}
                        to="/"
            >
                        Balance
            </NavLink>
</li>,
          <li>
            <NavLink
                        activeClassName="active"
                        aria-current="page"
                        to="/gastos"
            >
                        Gastos
            </NavLink>
</li>,
          <li>
            <NavLink
                        activeClassName="active"
                        aria-current="page"
                        to="/ingresos"
            >
                        Ingresos
            </NavLink>
</li>,
          <li>
            <NavLink
                        activeClassName="active"
                        aria-current="page"
                        to="#about"
            >
                        About
            </NavLink>
</li>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              aria-current="page"
              exact={true}
              to="/"
>
              Balance
</NavLink>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "Balance",
              "exact": true,
              "to": "/",
            },
            "ref": null,
            "rendered": "Balance",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              aria-current="page"
              to="/gastos"
>
              Gastos
</NavLink>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "Gastos",
              "to": "/gastos",
            },
            "ref": null,
            "rendered": "Gastos",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              aria-current="page"
              to="/ingresos"
>
              Ingresos
</NavLink>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "Ingresos",
              "to": "/ingresos",
            },
            "ref": null,
            "rendered": "Ingresos",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              aria-current="page"
              to="#about"
>
              About
</NavLink>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "About",
              "to": "#about",
            },
            "ref": null,
            "rendered": "About",
            "type": [Function],
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul>
          <li>
                    <NavLink
                              activeClassName="active"
                              aria-current="page"
                              exact={true}
                              to="/"
                    >
                              Balance
                    </NavLink>
          </li>
          <li>
                    <NavLink
                              activeClassName="active"
                              aria-current="page"
                              to="/gastos"
                    >
                              Gastos
                    </NavLink>
          </li>
          <li>
                    <NavLink
                              activeClassName="active"
                              aria-current="page"
                              to="/ingresos"
                    >
                              Ingresos
                    </NavLink>
          </li>
          <li>
                    <NavLink
                              activeClassName="active"
                              aria-current="page"
                              to="#about"
                    >
                              About
                    </NavLink>
          </li>
</ul>,
        "className": "bar",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <NavLink
                            activeClassName="active"
                            aria-current="page"
                            exact={true}
                            to="/"
              >
                            Balance
              </NavLink>
</li>,
            <li>
              <NavLink
                            activeClassName="active"
                            aria-current="page"
                            to="/gastos"
              >
                            Gastos
              </NavLink>
</li>,
            <li>
              <NavLink
                            activeClassName="active"
                            aria-current="page"
                            to="/ingresos"
              >
                            Ingresos
              </NavLink>
</li>,
            <li>
              <NavLink
                            activeClassName="active"
                            aria-current="page"
                            to="#about"
              >
                            About
              </NavLink>
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                aria-current="page"
                exact={true}
                to="/"
>
                Balance
</NavLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Balance",
                "exact": true,
                "to": "/",
              },
              "ref": null,
              "rendered": "Balance",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                aria-current="page"
                to="/gastos"
>
                Gastos
</NavLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Gastos",
                "to": "/gastos",
              },
              "ref": null,
              "rendered": "Gastos",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                aria-current="page"
                to="/ingresos"
>
                Ingresos
</NavLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Ingresos",
                "to": "/ingresos",
              },
              "ref": null,
              "rendered": "Ingresos",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                aria-current="page"
                to="#about"
>
                About
</NavLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "About",
                "to": "#about",
              },
              "ref": null,
              "rendered": "About",
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
